index.js

```diff
@@ -56,8 +61,10 @@ export const isElement =
    * When a `parent` node is known the `index` of node should also be given.
    *
    * @type {(
-   *   (<T extends Element>(node: unknown, test: T['tagName']|TestFunctionPredicate<T>|Array.<T['tagName']|TestFunctionPredicate<T>>, index?: number, parent?: Parent, context?: unknown) => node is T) &
-   *   ((node?: unknown, test?: Test, index?: number, parent?: Parent, context?: unknown) => boolean)
+   *   (() => false) &
+   *   (<T extends Element>(node: unknown, test: PredicateTest<T>, index?: number, parent?: Parent, context?: unknown) => node is T) &
+   *   ((node: unknown, test?: PredicateTest<Element>, index?: number, parent?: Parent, context?: unknown) => node is Element) &
+   *   ((node: unknown, test: Test, index?: number, parent?: Parent, context?: unknown) => boolean)
```

Comment:
The types are getting a bit complex, especially if we want to start inferring things.
It might be useful to move some types to a separate file, e.g. like https://github.com/syntax-tree/unist-util-visit-parents/blob/18d36ad5b14a15da0deb02f87d2b0e162e60d31e/complex-types.d.ts

Reply:
I tried to extract all related types to `complex-types.d.ts` and `@typedef {import('./complex-types'). PredicateTest} PredicateTest` errors with `PredicateTest` requires a generic parameter, not sure how to fix it. Maybe leave it as-is for now, and extract it in another PR, or you can help to extract later.

Reply:
yeah, that sucks. You have to pass generics at the place of import.

But, you don’t have to import-and-assign in a `@typedef`! See: https://github.com/syntax-tree/unist-util-visit-parents/blob/18d36ad5b14a15da0deb02f87d2b0e162e60d31e/index.js#L39-L40

Reply:
After https://github.com/syntax-tree/hast-util-is-element/pull/3#discussion_r760043742, I figured out the added overload is not needed anymore.

There is only one more `() => false` added now, so I think it is still OK to keep it as-is.

Reply:
Is there anything blocked left?